paymentapi/src/test/java/fi/hel/verkkokauppa/payment/api/PaytrailPaymentControllerTest.java

```diff
@@ -0,0 +1,200 @@
+import fi.hel.verkkokauppa.payment.repository.PaymentItemRepository;
+import fi.hel.verkkokauppa.payment.repository.PaymentRepository;
+import fi.hel.verkkokauppa.payment.testing.annotations.RunIfProfile;
+import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+
+import lombok.extern.slf4j.Slf4j;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@Slf4j
+public class PaytrailPaymentControllerTest {
+
+    @Autowired
+    private PaytrailPaymentController paytrailPaymentController;
+
+    @Autowired
+    private PaymentRepository paymentRepository;
+
+    @Autowired
+    private PayerRepository payerRepository;
+
+    @Autowired
+    private PaymentItemRepository paymentItemRepository;
+
+    private ArrayList<String> toBeDeletedPaymentId = new ArrayList<>();
+
+    @After
+    public void tearDown() {
+        try {
+            toBeDeletedPaymentId.forEach(paymentId -> {
+                log.info(paymentId);
+                paymentRepository.deleteById(paymentId);
+                payerRepository.deleteByPaymentId(paymentId);
+                paymentItemRepository.deleteByPaymentId(paymentId);
+            });
+            // Clear list because all deleted
+            toBeDeletedPaymentId = new ArrayList<>();
+        } catch (Exception e) {
+            log.info("delete error {}", e.toString());
+            toBeDeletedPaymentId = new ArrayList<>();
+        }
+    }
+
+    @Test
+    @RunIfProfile(profile = "local")
+    public void testCreatePaymentFromOrder() {
+        GetPaymentRequestDataDto paymentRequestDataDto = new GetPaymentRequestDataDto();
+        paymentRequestDataDto.setMerchantId("01fde0e9-82b2-4846-acc0-94291625192b");
+        OrderWrapper orderWrapper = createDummyOrderWrapper();
+        OrderItemDto dummyOrderItem = orderWrapper.getItems().get(0);
+        paymentRequestDataDto.setOrder(orderWrapper);
+
+        ResponseEntity<Payment> paymentResponse = paytrailPaymentController.createPaymentFromOrder(paymentRequestDataDto);
+        Payment payment = paymentResponse.getBody();
+
+        if (payment.getPaymentId() != null) {
+            /* Check payment */
+            assertEquals(orderWrapper.getOrder().getOrderId(), payment.getOrderId());
+
```

Comment:
Could you add assert for paytrailTransactionId is not null when payment is created?

Reply:
Yes good point, I'll add that